graphdatascience/procedure_surface/api/wcc_endpoints.py

```diff
@@ -0,0 +1,276 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, List, Optional
+
+from pandas import DataFrame
+
+from ...graph.graph_object import Graph
+
+
+class WccEndpoints(ABC):
+    """
+    Abstract base class defining the API for the Weakly Connected Components (WCC) algorithm.
+    """
+
+    @abstractmethod
+    def mutate(
+        self,
+        G: Graph,
+        mutate_property: str,
+        threshold: Optional[float] = None,
+        relationship_types: Optional[List[str]] = None,
+        node_labels: Optional[List[str]] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+        concurrency: Optional[Any] = None,
+        job_id: Optional[Any] = None,
+        seed_property: Optional[str] = None,
+        consecutive_ids: Optional[bool] = None,
+        relationship_weight_property: Optional[str] = None,
+    ) -> WccMutateResult:
+        """
+        Executes the WCC algorithm and writes the results to the in-memory graph as node properties.
+
+        Parameters
+        ----------
+        G : Graph
+            The graph to run the algorithm on
+        mutate_property : str
+            The property name to store the component ID for each node
+        threshold : Optional[float], default=None
+            The minimum required weight to consider a relationship during traversal
+        relationship_types : Optional[List[str]], default=None
+            The relationship types to project
+        node_labels : Optional[List[str]], default=None
+            The node labels to project
+        sudo : Optional[bool], default=None
+            Run analysis with admin permission
+        log_progress : Optional[bool], default=None
+            Whether to log progress
+        username : Optional[str], default=None
+            The username to attribute the procedure run to
+        concurrency : Optional[Any], default=None
+            The number of concurrent threads
+        job_id : Optional[Any], default=None
+            An identifier for the job
+        seed_property : Optional[str], default=None
+            Defines node properties that are used as initial component identifiers
+        consecutive_ids : Optional[bool], default=None
+            Flag to decide whether component identifiers are mapped into a consecutive id space
+        relationship_weight_property : Optional[str], default=None
+            The property name that contains weight
+
+        Returns
+        -------
+        WccMutateResult
+            Algorithm metrics and statistics
+        """
+        pass
```

Comment:
I suppose this is the "extended documentation burden" that I was highlighting in the ad-hoc discussion. It isn't nothing, but it is co-located in code and requires no additional tooling over the sphinx setup that we already have.

While we will still need to do non-trivial changes to the asciidoc parts of the client manual to align its content with the new API design, I do agree that the majority of change is contained to things like this.